### DISCUSSION TITLE
closing the server-client connection after certain time to limit data usage

### DISCUSSION THREAD

**Discussion by pddpauw** [mcguirepr89/BirdNET-Pi#1120] [Q&A]
Hi all,

I have a setup with Pi's connected to 4G dongles, and they are open to the public (peer birders helping with observations).

All works well, but was suprised on the 4G data usage.

My hypothesis is that the connection between client and server (=thus the Pi) stays open until the browser window gets closed by the client. The updated info gets sent to the client (even if not actively used by client).

To limit this data usage, I would like to close the connection after e.g. 5 minutes.  Is there anyone who has tried this, or would know where to start?

Thx, Pieter

**Comment by stevedee** [comment]
Are you sure its BirdNet?

I just installed NetHogs on my BirdNet-Pi system and ran it via ssh;
![NetHogs_BirdNet](https://github.com/mcguirepr89/BirdNET-Pi/assets/1870807/1575e272-024a-41db-a2b1-f1a7c0fdd6b6)


...and if I close the remote browser BirdNet-Pi tab;
![NetHogs_noClient](https://github.com/mcguirepr89/BirdNET-Pi/assets/1870807/76336c2f-cbee-4d06-9179-e825351776ef)

**Reply by pddpauw** [reply]
not sure I understand your reply...  the only thing the dongle is transmitting, is from and to the Pi.  Only other thing running on the Pi, is a VPNclient.
Think your log is confirming what I wrote: client browser closing => data transmission going down.  But happy to hear thoughts!

**Reply by stevedee** [reply]
Sorry Pieter, when you said you were surprised on the 4G data usage, I assumed it was very high.
At 0.135kB/sec its using less than 500kB/hr, but maybe you have lots of users.
Anyway, I'm looking into it.

**Reply by pddpauw** [reply]
ok, got your point ☺️! Will for sure look into this over the weekend.  Think we have now for one day (and one Pi) 400MB of data, which is a lot! Some users might have checked a couple of wav files, but like your suggestion of the Nethoghs tool!

**Reply by pddpauw** [reply]
OK, so, looked into this...
The issue is that in the overview.php, there are loops which send data as .png file.  Example of this, is the spectrogram.png wich is approx 300Kbyte in my case.  With this set-up in a nature reserve, I have a recording (and thus updated png file) every 15 secs, thus 1.2Mbyte per minute, or easily 30Mbyte per day and per connection.
On top, several other frames get renewed every second, which adds to this.

I've reworked this part of the code to remove the  automatic refresh (the setInterval() functions).  A refresh needs to be triggered with a refresh of the browser window (which is definitely OK for our purpose).
